db/database.go

```diff
@@ -834,6 +834,7 @@ func (db *Database) ForEachDocID(callback ForEachDocIDFunc, resultsOpts ForEachD
 // Iterate over the results of an AllDocs query, performing ForEachDocID handling for each row
 func (db *Database) processForEachDocIDResults(callback ForEachDocIDFunc, limit uint64, results sgbucket.QueryResultIterator) error {
```

Comment:
What do you think about making the `gocbRawIterator.Close()` implementation drain any remaining results prior to closing?

That covers use-cases outside of this `all_docs` one, and transparently does what the SDK needs to do in order to close a result iterator to keep the API consistent with the others (no special cases for GSI and gocbv1/v2 needed).

Reply:
Agree this would probably be best. However at the moment we can't do this.
See this which has a description of the issue: https://issues.couchbase.com/browse/GOCBC-1169

Reply:
@JRascagneres What's our use case that 'has multiple places where we call NextBytes()', that you mention on that ticket?

Reply:
Well just anywhere we call `Next()` on a `sgbucket.QueryResultIterator` we ultimately call `NextBytes()`. The issue we were hitting with all docs is that we weren't exhausting all of the results before calling close. This PR added that fix to exhaust the results on an all_docs request, however, as Ben pointed out this would be better done as part of our `Close()` wrapper to ensure this works for all uses and future uses of iterating on a query set.